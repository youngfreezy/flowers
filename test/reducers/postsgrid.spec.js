import reducer from 'reducers/posts';
import { ActionTypes } from 'constants/index';

describe('PostsGrid', () => {
  it('should return the initial state', () => {
    expect(reducer.posts(undefined, {})).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.GET_POSTS}`, () => {
    expect(
      reducer.posts(undefined, {
        type: ActionTypes.GET_POSTS,
        payload: { q: 'react' },
      }),
    ).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.GET_POSTS_SUCCESS}`, () => {
    expect(
      reducer.posts(undefined, {
        type: ActionTypes.GET_POSTS_SUCCESS,
        payload: {},
      }),
    ).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.GET_POSTS_FAILURE}`, () => {
    expect(
      reducer.posts(undefined, {
        type: ActionTypes.GET_POSTS_FAILURE,
        payload: {},
      }),
    ).toMatchSnapshot();
  });
});
