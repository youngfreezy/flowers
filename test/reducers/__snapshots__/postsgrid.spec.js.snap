// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsGrid should handle GET_POSTS 1`] = `
Object {
  "items": Array [],
  "status": "idle",
}
`;

exports[`PostsGrid should handle GET_POSTS_FAILURE 1`] = `
Object {
  "items": "Something went wrong",
  "status": "error",
}
`;

exports[`PostsGrid should handle GET_POSTS_SUCCESS 1`] = `
Object {
  "items": Array [],
  "status": "success",
}
`;

exports[`PostsGrid should return the initial state 1`] = `
Object {
  "items": Array [],
  "status": "idle",
}
`;
